spec/yoti/data_type/document_details_spec.rb

```diff
@@ -0,0 +1,138 @@
+      it 'should return 2020-01-01' do
+        expect(document_details.expiration_date).to be_a_kind_of(DateTime)
+        expect(document_details.expiration_date.to_s).to eql('2020-01-01T00:00:00+00:00')
+      end
+    end
+  end
+
+  context 'when there are no optional attributes' do
+    let :document_details do
+      Yoti::DocumentDetails.new('DRIVING_LICENCE GBR 1234abc')
+    end
+    describe '.type' do
+      it 'should return DRIVING_LICENCE' do
+        expect(document_details.type).to eql 'DRIVING_LICENCE'
+      end
+    end
+    describe '.issuing_country' do
+      it 'should return GBR' do
+        expect(document_details.issuing_country).to eql 'GBR'
+      end
+    end
+    describe '.document_number' do
+      it 'should return 1234abc' do
+        expect(document_details.document_number).to eql('1234abc')
+      end
+    end
+    describe '.expiration_date' do
+      it 'should return nil' do
+        expect(document_details.expiration_date).to be_nil
+      end
+    end
+  end
+
+  context 'when there are two optional attributes' do
+    document_details = Yoti::DocumentDetails.new('DRIVING_LICENCE GBR 1234abc 2016-05-01 DVLA')
+    describe '.type' do
+      it 'should return DRIVING_LICENCE' do
+        expect(document_details.type).to eql('DRIVING_LICENCE')
+      end
+    end
+    describe '.issuing_country' do
+      it 'should return GBR' do
+        expect(document_details.issuing_country).to eql('GBR')
+      end
+    end
+    describe '.document_number' do
+      it 'should return 1234abc' do
+        expect(document_details.document_number).to eql('1234abc')
+      end
+    end
+    describe '.expiration_date' do
+      it 'should return 2016-05-01' do
+        expect(document_details.expiration_date).to be_a_kind_of(DateTime)
+        expect(document_details.expiration_date.to_s).to eql('2016-05-01T00:00:00+00:00')
+      end
+    end
+    describe '.issuing_authority' do
+      it 'should return DVLA' do
+        expect(document_details.issuing_authority).to eql('DVLA')
+      end
+    end
+  end
+
+  context 'when value is empty' do
+    it 'should raise ArgumentError' do
+      expect { Yoti::DocumentDetails.new('') }
+        .to raise_error(ArgumentError, 'Invalid value for Yoti::DocumentDetails')
+    end
+  end
+
+  context 'when the country is invalid' do
+    it 'should raise ArgumentError' do
+      expect { Yoti::DocumentDetails.new('PASSPORT 13 1234abc 2016-05-01') }
+        .to raise_error(ArgumentError, 'Invalid value for Yoti::DocumentDetails')
+    end
+  end
+
+  context 'when the document number is invalid' do
+    it 'should raise ArgumentError' do
+      expect { Yoti::DocumentDetails.new('PASSPORT GBR $%^$%^£ 2016-05-01') }
+        .to raise_error(ArgumentError, 'Invalid value for Yoti::DocumentDetails')
+    end
+  end
+
+  context 'when the expiration date is missing' do
+    document_details = Yoti::DocumentDetails.new('PASS_CARD GBR 22719564893 - CITIZENCARD')
```

Comment:
Do you think we need a test that checks that something after the `-` is parsed, or do the current tests cover that?

Reply:
Added in a311b17